primitives/runtime/src/lib.rs

```diff
@@ -507,6 +507,24 @@ impl WeightExtraOps for Weight {
 	}
 }
 
+/// Trait that provides a static `str`.
+pub trait StaticStrProvider {
+	const STR: &'static str;
+}
+
+#[macro_export]
+macro_rules! generate_static_str_provider {
+	($str:expr) => {
+		$crate::paste::item! {
+			pub struct [<Str $str>];
```

Comment:
It's in case we want to use the same name for the signed extension. For example something like:

```
bp_runtime::generate_static_str_provider!(BridgeRefundRialtoPara2000Lane0Msgs);
/// Signed extension that refunds relayers that are delivering messages from the Rialto parachain.
pub type BridgeRefundRialtoPara2000Lane0Msgs = RefundBridgedParachainMessages<
	Runtime,
	RefundableParachain<WithRialtoParachainsInstance, RialtoParachainId>,
	RefundableMessagesLane<WithRialtoParachainMessagesInstance, RialtoParachainMessagesLane>,
	ActualFeeRefund<Runtime>,
	StrBridgeRefundRialtoPara2000Lane0Msgs,
>;
```

Reply:
Yes, but then the id of the signed extension would also be `StrBridgeRefundRialtoPara2000Lane0Msgs`. I think it would be better for the id to be just `BridgeRefundRialtoPara2000Lane0Msgs`.

Anyway another option would also be to change the macro in order to accept 2 parameters. The name of the struct and the string. I can do it if you think it's needed.

Reply:
ok, ok, not needed :), I didnt realize that, I got it now: `[<Str $str>]` vs `stringify!($str)`